dsn.go

```diff
@@ -8,7 +8,7 @@ import (
 
 var (
 	// Regexp syntax: https://github.com/google/re2/wiki/Syntax
-	reDSN   = regexp.MustCompile(`^([a-zA-Z0-9_-]+):([a-zA-Z0-9_-]+)@([a-zA-Z0-9/_.-]+)\?([^/]+)$`)
+	reDSN   = regexp.MustCompile(`^([a-zA-Z0-9_-]+):([=a-zA-Z0-9_-]+)@([:a-zA-Z0-9/_.-]+)\?([^/]+)$`)
```

Comment:
Seems endpoint part do not need `:`?

Reply:
endpoints such as: `maxcompute.server.com:8080` or a test env `10.1.2.3:8002`

Reply:
I see!